baybe/recommenders/pure/bayesian/base.py

```diff
@@ -28,14 +28,14 @@ class BayesianRecommender(PureRecommender, ABC):
     """The used surrogate model."""
 
     acquisition_function: AcquisitionFunction = field(
-        converter=convert_acqf, factory=qExpectedImprovement, kw_only=True
+        converter=convert_acqf, factory=qExpectedImprovement
```

Comment:
Not "necessary" but I noticed that the signature was not very nice:
<img width="429" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/fb3dc8df-19a7-45b2-aa9a-06d56af01627">
The logical order to me is (ignoring the deprecated argument): 1) Surrogate 2) ACQF 3) Rest

And this is indeed what you get after the change:
<img width="430" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/c2a1acef-b012-400a-b964-a7cbc255af4b">

We can discuss though whether we want `sequential` as `kwonly` or not. Right now, I've made it keyword. Preferences?

Reply:
ok seems alright

on that not: why is `acquisition_function_cls` a `bool` and why is it not `Optional` ? I know that was not introuced in this PR and its the deprecated thing, but that still seems wrong?

Reply:
Well, you may ask that yourself :D Here the changes from the commit you made in the acqf refactoring:
<img width="561" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/73a00f94-b168-4a3e-9fe4-2722a88fcc1b">
<img width="396" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/962033fa-58a9-4cb0-bc6c-3aa3d52962ae">
So I guess you copy-pasted the logic from the campaign deprecation I introduced
<img width="627" alt="image" src="https://github.com/emdgroup/baybe/assets/23265127/1026fb62-dabb-4baa-9b86-39cdeebc09d4">
but did not adjust the types, I and didn't bother during review since it's a deprecation anyway. But I'm happy to fix it. In fact, have just pushed a commit. Please confirm if my explanation was correct and resolve if happy